File: app/src/main/java/com/example/foodcouponautomation/MainActivity.java
package com.example.foodcouponautomation;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.KeyEvent;
import android.view.View;
import android.view.inputmethod.EditorInfo;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.AppCompatButton;

import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;
import com.google.android.material.textfield.TextInputEditText;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.firestore.FirebaseFirestore;
import com.google.firebase.firestore.QueryDocumentSnapshot;
import com.google.firebase.firestore.QuerySnapshot;

public class MainActivity extends AppCompatActivity {

    TextInputEditText editTextCountryCode, editTextPhone;
    AppCompatButton buttonContinue;
    private FirebaseFirestore db;
    private String TAG = MainActivity.class.getCanonicalName();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        db = FirebaseFirestore.getInstance();

        editTextCountryCode = findViewById(R.id.editTextCountryCode);
        editTextPhone = findViewById(R.id.editTextPhone);
        buttonContinue = findViewById(R.id.buttonContinue);

        editTextPhone.setOnEditorActionListener(new EditText.OnEditorActionListener() {

            @Override
            public boolean onEditorAction(TextView v, int actionId,
                                          KeyEvent event) {
                if (event != null && event.getKeyCode() == KeyEvent.KEYCODE_ENTER
                        || actionId == EditorInfo.IME_ACTION_DONE) {
                    onActionPerformed();
                    return true;
                }
                return false;
            }
        });

        buttonContinue.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
              onActionPerformed();
            }
        });
    }

    private void onActionPerformed() {
        String number = editTextPhone.getText().toString().trim();

        if (number.isEmpty() || number.length() != 10) {
            editTextPhone.setError("Valid number is required");
            editTextPhone.requestFocus();
            return;
        }

        checkUserMobileNoPresent(number);
    }

    private void checkUserMobileNoPresent(final String phoneNo) {

        db.collection("users").whereEqualTo("mob", phoneNo).get()
                .addOnCompleteListener(new OnCompleteListener<QuerySnapshot>() {
                    @Override
                    public void onComplete(@NonNull Task<QuerySnapshot> task) {
                        if (task.isSuccessful()) {
                            boolean isMobileNoPresent = false;
                            for (QueryDocumentSnapshot document : task.getResult()) {
                                Log.i(TAG, document.getId() + " => " + document.getData());
                                isMobileNoPresent = true;
                            }

                            if (isMobileNoPresent) {
                                String code = editTextCountryCode.getText().toString().trim();
                                String phoneNumber = code + phoneNo;
                                Intent intent = new Intent(MainActivity.this, VerifyPhoneActivity.class);
                                intent.putExtra("phoneNumber", phoneNumber);
                                startActivity(intent);
                            } else {
                                editTextPhone.setError("Your number is not present");
                                editTextPhone.requestFocus();
                            }

                        } else {
                            Toast.makeText(getApplicationContext(), "Something went wrong", Toast.LENGTH_LONG);
                            Log.e(TAG, "Error getting documents: ", task.getException());
                        }
                    }
                });
    }

    @Override
    protected void onStart() {
        super.onStart();

        if (FirebaseAuth.getInstance().getCurrentUser() != null) {
            Intent intent = new Intent(this, ProfileActivity.class);
            intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);

            startActivity(intent);
        }
    }
}

File: app/src/main/java/com/example/foodcouponautomation/MainActivity.kt
package com.example.foodcouponautomation

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import kotlinx.android.synthetic.main.activity_main.*
import com.google.firebase.firestore.FirebaseFirestore



class MainActivity : AppCompatActivity() {

    var TAG = MainActivity::class.java.canonicalName
    val db = FirebaseFirestore.getInstance()
    var isValidMobileNumber = false
    var list:List<User>  = ArrayList<User>()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


        db.collection("users")
            .get()
            .addOnCompleteListener {
                    list ->
                if(list.isSuccessful){

                    var documents = list.result?.documents

                    Log.i(TAG,"vdshcvh");
                    Log.i(TAG, list.result?.documents?.size.toString());
                    Log.i(TAG, list.result?.documents?.get(0)?.id);
                    Log.i(TAG, list.result?.documents?.get(0)?.get("mob")?.toString());

                    isValidMobileNumber = list.result?.documents?.get(0)?.get("mob")?.toString().equals(etMobileNumber.text.toString())

                }else{
                    Log.i(TAG,"cancel");
                }

            }


        submitButton.setOnClickListener {

            if(isUserAuthenticated()){
                startActivity(Intent(this,CouponsActivity::class.java))
            }

        }





    }

    private fun isUserAuthenticated(): Boolean {


        Log.i(TAG, isValidMobileNumber.toString());
        Log.i(TAG, etMobileNumber.text.toString());


        return isValidMobileNumber
    }
}
